lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -1250,13 +1266,23 @@ func removeAnonContainerVols(pl *client.PortLayer, cID string, volumes map[strin
 
 	for vol := range volumes {
 		// Extract the volume ID from the full mount path, which is of form "id:mountpath:flags" - see getMountString().
-		volFields := strings.SplitN(vol, ":", 2)
+		volFields := strings.SplitN(vol, ":", 3)
+
+		// XXX: this check will start to fail when we fix our metadata correctness issues
```

Comment:
Instead of an `XXX:` we should perhaps use a `NOTE(<name>):` so it's easier to address in the future (I'm not sure what an `XXX` denotes here).